src/Processor.php

```diff
@@ -143,4 +151,68 @@ private function readFromFile()
         $reader = new Reader();
         $this->data = $reader->fromFile($path);
     }
+
+    private function getVaultVariables()
+    {
+        $this->vaultVariables = $this->filterMultiArray($this->data);
+    }
+
+    public function filterMultiArray(array $data)
+    {
+        $filteredArray = [];
+        foreach ($data as $value) {
+            if (is_array($value)) {
+                $filteredSubArray = $this->filterMultiArray($value);
+                $filteredArray = array_merge($filteredArray, $filteredSubArray);
+            } elseif (str_contains($value, self::VAULT_KEYWORD)) {
+                $filteredArray[] = $value;
+            }
+        }
+
+        return $filteredArray;
+    }
+
+    private function sortVariablesByRouteAndKey()
+    {
+        //todo maybe remove strtolower when variables are stored in apper case in vault.
+        $sorted = [];
+        foreach ($this->vaultVariables as $vaultVariable) {
+            $array = explode('/', strtolower($vaultVariable));
+            $secretKey = array_pop($array);
+            array_shift($array);
+            $secretRoute = implode('/', $array);
+            $sorted[$secretRoute][] = $secretKey;
+        }
+
+        return $sorted;
+    }
+
+    private function getValuesFromVault()
+    {
+        $sorted = $this->sortVariablesByRouteAndKey();
+
+        $data = [];
+        $repo = new VaultRepository($this->data[self::VAULT_KEYWORD]);
```

Comment:
VaultRepository should be set via a setter

```
/** @var VaultRepository */
private $vaultRepository;

public function setVaultRepository(VaultRepository $vaultRepository)
{
    $this->vaultRepository = $vaultRepository;
}
```

Also, in this method getValuesFromVault() check if 
```
if ($this->vaultRepository === null) {
    throw new \RuntimeException("VaultRepository not set. Use setVaultRepository() to set it");
}
```